rotkehlchen/globaldb/handler.py

```diff
@@ -770,11 +790,18 @@ def check_asset_exists(
         If it exists it returns a list of the identifiers of the assets.
         """
         cursor = GlobalDBHandler().conn.cursor()
-        query = cursor.execute(
-            'SELECT A.identifier from assets AS A JOIN common_asset_details as C '
-            'ON A.identifier=C.identifier WHERE A.type=? AND A.name=? AND C.symbol=?;',
-            (asset_type.serialize_for_db(), name, symbol),
-        )
+        if asset_type == AssetType.CUSTOM_ASSET:
+            query = cursor.execute(
+                'SELECT identifier from assets WHERE type=? AND name=?;',
+                (AssetType.CUSTOM_ASSET.serialize_for_db(), name),
+            )
+        else:
+            query = cursor.execute(
+                'SELECT A.identifier from assets AS A JOIN common_asset_details as C '
+                'ON A.identifier=C.identifier WHERE A.type=? AND A.name=? AND C.symbol=?;',
+                (asset_type.serialize_for_db(), name, symbol),
+            )
```

Comment:
If you check where this is used at the moment of importing assets we can just use `add_custom_asset` that has `_raise_if_custom_asset_exists`

Reply:
I would not use `add_custom_asset` since it requires user db handler, and here we only have global db handler

Reply:
But you shouldn't use it here. Instead of modifying this functions you should modify `import_assets_from_file` and it will be cleaner. Please check that function